homeassistant/components/auth/__init__.py

```diff
@@ -180,9 +181,11 @@
 
 
 @bind_hass
-def create_auth_code(hass, client_id: str, user: User) -> str:
+def create_auth_code(
+    hass, client_id: str, credential_or_user: Union[Credentials, User]
```

Comment:
Can we just make this always `Credentials` ?

Reply:
If we remove the ability to create an auth code directly for the user we can change this. There is quite some infrastructure to support creating auth codes for different types of objects. I think it should match the potential parameters of the underlying function being called.

I can create a separate pull to drop that extra code if we deem it not needed.